web/modules/custom/dpl_recommender/src/Controller/DplRecommenderController.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Drupal\dpl_recommender\Controller;
+
+use Drupal\Core\Block\BlockManagerInterface;
+use Drupal\Core\Controller\ControllerBase;
+use Drupal\Core\Render\RendererInterface;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+
+/**
+ * Render recommender react app.
+ */
+class DplRecommenderController extends ControllerBase {
+
+  /**
+   * DplRecommenderController constructor.
+   *
+   * @param \Drupal\Core\Block\BlockManagerInterface $blockManager
+   *   Drupal block manager.
+   * @param \Drupal\Core\Render\RendererInterface $renderer
+   *   Drupal renderer service.
+   */
+  public function __construct(
+    private BlockManagerInterface $blockManager,
+    private RendererInterface $renderer
+  ) {
+  }
+
+  /**
+   * {@inheritdoc}
+   *
+   * @param \Symfony\Component\DependencyInjection\ContainerInterface $container
+   *   The Drupal service container.
+   *
+   * @return static
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('plugin.manager.block'),
+      $container->get('renderer'),
+    );
+  }
+
+  /**
+   * Display page with the recommender component.
+   *
+   * @return mixed[]
+   *   Render array.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginException
+   */
+  public function list(): array {
+    $config = [];
+
+    /** @var \Drupal\dpl_recommender\Plugin\Block\RecommenderBlock $plugin_block */
+    $plugin_block = $this->blockManager->createInstance('dpl_recommender_block', $config);
+
+    $access_result = $plugin_block->access($this->currentUser());
+    if (is_object($access_result) && $access_result->isForbidden() || is_bool($access_result) && !$access_result) {
+      throw new AccessDeniedHttpException();
+    }
```

Comment:
I have a feeling that this is going to be written repeatedly (if it isn't already). Could we define a service or a global function for checking access?

Reply:
Fine idea, but think we should wait to all have been merged and then refactor (will make it easier to do as right now it's an mix).

Reply:
Ok. I wrote this in other reviews. I'll shut up about it and look forward to have it refactored.
I think it would be a good idea to put in a TODO then.